packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardFilterBuilder.tsx

```diff
@@ -438,21 +458,34 @@ export const CloudPulseDashboardFilterBuilder = React.memo(
             <Typography variant="h3">Filters</Typography>
           </Button>
         </GridLegacy>
-        <GridLegacy
-          columnSpacing={theme.spacing(2)}
-          container
-          item
-          sx={{
-            display: showFilter ? 'flex' : 'none',
-            maxHeight: theme.spacing(23),
-            overflow: 'auto',
-            pr: { sm: 0, xs: 2 },
-            rowGap: theme.spacing(2),
-          }}
-          xs={12}
-        >
-          <RenderFilters />
-        </GridLegacy>
+        {isResourceCallLoading ? (
+          <GridLegacy
+            alignItems="center"
+            container
+            display="flex"
+            justifyContent="center"
+          >
+            <GridLegacy item>
+              <CircleProgress size="md" />
+            </GridLegacy>
```

Comment:
```suggestion
              <CircleProgress size="md" />
```

grid item is not required

Reply:
Let me see as well